flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/PartitionRequestClient.java

```diff
@@ -171,10 +171,7 @@ public void operationComplete(ChannelFuture future) throws Exception {
 	}
 
 	public void notifyCreditAvailable(RemoteInputChannel inputChannel) {
-		// We should skip the notification if the client is already closed.
-		if (!closeReferenceCounter.isDisposed()) {
-			clientHandler.notifyCreditAvailable(inputChannel);
-		}
+		clientHandler.notifyCreditAvailable(inputChannel);
```

Comment:
It seems to me like this `if` check was not protecting us from any race condition, there was no other synchronisation between `#notifyCreditAvailable` and `#close()` - `notifyCreditAvailable` could execute this check first and freeze, then `close()` is called, disposes everything and exits and after that `notifyCreditAvailable` could resume working.

On the other hand, this `if` check was saving us some work only in rare case of disposing channels in an exchange of synchronisation cost on common path.

Am I right?